hardware/awg/tektronix_awg70k.py

```diff
@@ -116,6 +128,18 @@ def on_activate(self):
         self.__max_seq_repetitions = int(self.query('SLIS:SEQ:STEP:RCO:MAX?'))
         self.__min_waveform_length = int(self.query('WLIS:WAV:LMIN?'))
         self.__max_waveform_length = int(self.query('WLIS:WAV:LMAX?'))
+        self.__has_sequence_mode = '03' in self.query('*OPT?').split(',')
+
+        # CW microwave attributes
+        self.__mw_mode = MicrowaveMode.CW
+        self.__mw_cw_params = {'freq': 2.87e9, 'power': -30}
+        self.__mw_sweep_params = {'start': 2.82e9, 'stop': 2.92e9, 'step': 2e6, 'power': -30}
+        self.__mw_list_params = {'freq': list(np.linspace(2.82e9, 2.92e9, 101)), 'power': -30}
```

Comment:
Why set these parameters in on_activate and __init__ twice?

Reply:
I just made it a habit to declare all instance attributes in `__init__` and initialize them first as default/empty data type (empty list/dict or `None` for more complex types). The "proper" initialization of these attributes follows in `on_activate`. 
But this way you can see all available variables easily at once. 
Additionally it satisfies code inspection tools since you should avoid declaring instance variables outside of `__init__`.
I could stick to my usual pattern and just initialize them as empty dicts in `__init__`.